library/spiders/books.py

```diff
@@ -0,0 +1,87 @@
+import scrapy
+from scrapy import Selector
+from scrapy.http import Response
+from selenium import webdriver
+from selenium.webdriver.chrome.options import Options
+from selenium.webdriver.common.by import By
+
+
+class BooksSpider(scrapy.Spider):
+    name = "books"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    def __init__(self):
+        super().__init__()
+        self._driver = self.__set_chrome_driver()
+        self._rating_table = {
+            "Zero": 0,
+            "One": 1,
+            "Two": 2,
+            "Three": 3,
+            "Four": 4,
+            "Five": 5
+        }
+
+    def __set_chrome_driver(self):
+        options = Options()
+        options.add_argument("--headless=new")
+        return webdriver.Chrome(options=options)
+
+    def close(self, reason: str):
+        self._driver.quit()
+
+    def parse(self, response: Response, **kwargs):
+        for product in response.css(".product_pod"):
+            yield self._parse_book_page(response, product)
+
+        next_page = response.css(".pager > .next").css("a::attr(href)").get()
+
+        if next_page:
+            yield response.follow(next_page, callback=self.parse)
+
+
+    def _parse_book_page(
+            self,
+            response: Response,
+            product: Selector
+    ) -> dict:
+
+        detail_url = response.urljoin(product.css("a::attr(href)").get())
+        self._driver.get(detail_url)
+        return self._get_book_data()
```

Comment:
The `_parse_book_page` method uses Selenium to navigate to each book's detail page. This approach can be slow and resource-intensive. If the data can be extracted using Scrapy alone, it would be more efficient.